core/container_create.go

```diff
@@ -47,6 +47,8 @@ func (ds *dockerService) CreateContainer(
 		return nil, fmt.Errorf("sandbox config is nil for container %q", config.Metadata.Name)
 	}
 
+	// TODO: Docker supports annotations for now, so labels and annotations might be separated
+	//       in the future. Keeping them merged is just for backward compatibility.
```

Comment:
Backward compatibility with what? AIUI no containers will be persisted across an upgrade (daemon restart) so there is no need for containers started by cri-dockerd v0.3.x to be understandable by v0.4.x, or vice versa.

Reply:
There are two main considerations for backward compatibility: 
- I‘m not sure if some runtimes use the label to make some judgments
- Docker ContainerList api doesn't return annotations, so keeping them merged is needed for later extraction.